### PR TITLE
Installation fixes

### PR DESCRIPTION
**Pull request by podsvirov**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -273,7 +273,10 @@ else()
   message(STATUS "Building libbinaryen as shared library.")
   add_library(binaryen SHARED ${binaryen_SOURCES} ${binaryen_objs})
 endif()
-install(TARGETS binaryen DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS binaryen
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
```
Comment (kripken) [inline CMakeLists.txt:279]: as this is a library, doesn't only the middle one here matter? I might be missing something...

Comment (podsvirov) [reply CMakeLists.txt:279]: On the Windows platform, shared libraries have a runtime component (* .dll file), which is usually located in the 'bin' directory (near executable programs).

Comment (sbc100) [reply CMakeLists.txt:279]: Hmm.. but that isn't true on non-windows platforms.   `.so` files should not live in the `bin` directory on other platforms.

I find it odd these this stuff isn't all just handled by default with cmake.

Comment (sbc100) [reply CMakeLists.txt:279]: The other parts of this change LGTM for sure so you might want split of this part ?

**Comment by podsvirov** [comment]
@sbc100 `.so` is a `LIBRARY` component on non-DLL platforms and it's will be placed to `lib` directory - all good.

**Comment by sbc100** [comment]
OK sounds reasonable.  Do you know what those three extra lines are needed?  What doesn't cmake just the right thing be default?  I would have  through that the behaviour we are asking for is nothing but the standard expected bahaviour.

**Comment by podsvirov** [comment]
Yes.  Yes.  You're right.  You can see these three lines in many projects.

**Comment by podsvirov** [comment]
What about to merge this PR?

**Comment by kripken** [comment]
Merging, thanks again @podsvirov !

**Pull request merged**